deepspeed/runtime/dataloader.py

```diff
@@ -68,9 +68,10 @@ def __init__(self,
         self.device_count = device_count
         self.batch_size = batch_size
         self.pin_memory = pin_memory
-        self.len = len(self.data_sampler)
-        self.data = None
         self.dataloader_drop_last = dataloader_drop_last
+        # create self.dataloader and self.data
+        self._create_dataloader()
```

Comment:
Why call create_dataloader() here? Our current approach is to create in __iter__. If I remember correctly, this is needed to handle multiple epoch. Did you encounter a problem with the current approach?